src/Microsoft.OData.Edm/Validation/ValidationRules.cs

```diff
@@ -1192,6 +1192,27 @@ public static class ValidationRules
                     }
                 });
 
+        // Add this to rule set as a breaking change for a future release.
+        /// <summary>
+        /// Validates that if a complex-typed property has a type that is the same as
+        /// declaring type of the complex type or a derived type, then the complex type is nullable.
+        /// </summary>
+        public static readonly ValidationRule<IEdmStructuralProperty> RecursiveComplexTypedPropertyMustBeOptional =
+            new ValidationRule<IEdmStructuralProperty>(
+                (context, property) =>
+                {
+                    IEdmTypeReference propType = property.Type;
+                    IEdmStructuredType declaredType = property.DeclaringType;
+
+                    if (propType.Definition == declaredType && !propType.IsNullable)

```

Comment:
do we care about the inheritance? 
For example:

A is base type, B is derived type.
Scenario:

1. A has A type property
2. A has B type proprety
3. B has A type property
4. B has B type property (same as 1)

Reply:
Expanded scope to include inheritance ✅